jsonnet/kube-prometheus/prometheus/prometheus.libsonnet

```diff
@@ -35,468 +24,438 @@ local k = import 'github.com/ksonnet/ksonnet-lib/ksonnet.beta.4/k.libsonnet';
-        metadata: {
-          name: 'kube-scheduler',
-          namespace: p.namespace,
-          labels: {
-            'k8s-app': 'kube-scheduler',
-          },
+    },
+
+    serviceMonitor: {
+      apiVersion: 'monitoring.coreos.com/v1',
+      kind: 'ServiceMonitor',
+      metadata: {
+        name: 'prometheus',
+        namespace: p.namespace,
+        labels: { 'k8s-app': 'prometheus' },
+      },
+      spec: {
+        selector: {
+          matchLabels: { prometheus: p.name },
         },
-        spec: {
-          jobLabel: 'k8s-app',
-          endpoints: [
-            {
-              port: 'https-metrics',
-              interval: '30s',
-              scheme: 'https',
-              bearerTokenFile: '/var/run/secrets/kubernetes.io/serviceaccount/token',
-              tlsConfig: {
-                insecureSkipVerify: true,
-              },
-            },
-          ],
-          selector: {
-            matchLabels: {
-              'k8s-app': 'kube-scheduler',
-            },
-          },
-          namespaceSelector: {
-            matchNames: [
-              'kube-system',
-            ],
-          },
+        endpoints: [{
+          port: 'web',
+          interval: '30s',
+        }],
+      },
+    },
+
+    serviceMonitorKubeScheduler: {
+      apiVersion: 'monitoring.coreos.com/v1',
+      kind: 'ServiceMonitor',
+      metadata: {
+        name: 'kube-scheduler',
+        namespace: p.namespace,
+        labels: { 'k8s-app': 'kube-scheduler' },
+      },
+      spec: {
+        jobLabel: 'k8s-app',
+        endpoints: [{
+          port: 'https-metrics',
+          interval: '30s',
+          scheme: 'https',
+          bearerTokenFile: '/var/run/secrets/kubernetes.io/serviceaccount/token',
+          tlsConfig: { insecureSkipVerify: true },
+        }],
+        selector: {
+          matchLabels: { 'k8s-app': 'kube-scheduler' },
+        },
+        namespaceSelector: {
+          matchNames: ['kube-system'],
         },
       },
-    serviceMonitorKubelet:
-      {
-        apiVersion: 'monitoring.coreos.com/v1',
-        kind: 'ServiceMonitor',
-        metadata: {
-          name: 'kubelet',
-          namespace: p.namespace,
-          labels: {
-            'k8s-app': 'kubelet',
+    },
+
+    serviceMonitorKubelet: {
+      apiVersion: 'monitoring.coreos.com/v1',
+      kind: 'ServiceMonitor',
+      metadata: {
+        name: 'kubelet',
+        namespace: p.namespace,
+        labels: { 'k8s-app': 'kubelet' },
+      },
+      spec: {
+        jobLabel: 'k8s-app',
+        endpoints: [
+          {
+            port: 'https-metrics',
+            scheme: 'https',
+            interval: '30s',
+            honorLabels: true,
+            tlsConfig: { insecureSkipVerify: true },
+            bearerTokenFile: '/var/run/secrets/kubernetes.io/serviceaccount/token',
+            metricRelabelings: (import 'kube-prometheus/dropping-deprecated-metrics-relabelings.libsonnet'),
```

Comment:
Would it be better to import this at the top of the file and use a variable here?